hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -315,6 +318,29 @@ public static Option<HoodieInstant> getEarliestInstantForMetadataArchival(
     }
   }
 
+  /**
+   * Validate user-specified timestamp of time travel query against incomplete commit's timestamp.
+   *
+   * @throws HoodieException when time travel query's timestamp >= incomplete commit's timestamp
+   */
+  public static void validateTimestampAsOf(HoodieTableMetaClient metaClient, String timestampAsOf) {
```

Comment:
are we not supporting the 3 enum values? looks like we are going w/ FAIL directly ?

Reply:
@nsivabalan thought that's what we agreed upon, as in only handle time travel by failing